test/e2e_node/pod_resize_test.go

```diff
@@ -0,0 +1,1768 @@
+						{"name":"c1", "resources":{"requests":{"memory":"400Mi"}}}
+					]}}`,
+			patchError: "Pod QoS is immutable",
+			expected: []e2epod.ResizableContainerInfo{
+				{
+					Name: "c1",
+				},
+				{
+					Name: "c2",
+				},
+				{
+					Name: "c3",
+				},
+			},
+		},
+	}
+
+	timeouts := framework.NewTimeoutContext()
+
+	for idx := range tests {
+		tc := tests[idx]
+		ginkgo.It(tc.name+policy.title+" (InPlacePodVerticalScalingExclusiveCPUs="+strconv.FormatBool(isInPlacePodVerticalScalingExclusiveCPUsEnabled)+")", func(ctx context.Context) {
+			podClient := e2epod.NewPodClient(f)
+			var testPod, patchedPod *v1.Pod
+			var pErr error
+
+			tStamp := strconv.Itoa(time.Now().Nanosecond())
+			e2epod.InitDefaultResizePolicy(tc.containers)
+			e2epod.InitDefaultResizePolicy(tc.expected)
+			testPod = e2epod.MakePodWithResizableContainers(f.Namespace.Name, "testpod", tStamp, tc.containers)
+			testPod = e2epod.MustMixinRestrictedPodSecurity(testPod)
+
+			ginkgo.By("creating pod")
+			newPod := podClient.CreateSync(ctx, testPod)
+
+			perr := e2epod.WaitForPodCondition(ctx, f.ClientSet, newPod.Namespace, newPod.Name, "Ready", timeouts.PodStartSlow, testutils.PodRunningReady)
+			framework.ExpectNoError(perr, "pod %s/%s did not go running", newPod.Namespace, newPod.Name)
+			framework.Logf("pod %s/%s running", newPod.Namespace, newPod.Name)
+
+			ginkgo.By("verifying initial pod resources, allocations, and policy are as expected")
+			e2epod.VerifyPodResources(newPod, tc.containers)
+			e2epod.VerifyPodResizePolicy(newPod, tc.containers)
+
+			ginkgo.By("verifying initial pod status resources and cgroup config are as expected")
+			e2epod.VerifyPodStatusResources(newPod, tc.containers)
+
+			ginkgo.By("patching pod for resize")
+			patchedPod, pErr = f.ClientSet.CoreV1().Pods(newPod.Namespace).Patch(ctx, newPod.Name,
+				types.StrategicMergePatchType, []byte(tc.patchString), metav1.PatchOptions{})
+			if tc.patchError == "" {
+				framework.ExpectNoError(pErr, "failed to patch pod for resize")
+			} else {
+				gomega.Expect(pErr).To(gomega.HaveOccurred(), tc.patchError)
+				patchedPod = newPod
+			}
+
+			ginkgo.By("verifying pod resources after patch")
+			e2epod.VerifyPodResources(patchedPod, tc.expected)
+
+			ginkgo.By("verifying pod allocations after patch")
+			gomega.Eventually(ctx, e2epod.VerifyPodAllocations, timeouts.PodStartShort, timeouts.Poll).
+				WithArguments(patchedPod, tc.expected).
+				Should(gomega.BeNil(), "failed to verify Pod allocations for patchedPod")
+
+			deletePodSyncByName(ctx, f, newPod.Name)
+			// we need to wait for all containers to really be gone so cpumanager reconcile loop will not rewrite the cpu_manager_state.
+			// this is in turn needed because we will have an unavoidable (in the current framework) race with the
+			// reconcile loop which will make our attempt to delete the state file and to restore the old config go haywire
+			waitForAllContainerRemoval(ctx, newPod.Name, newPod.Namespace)
```

Comment:
same comment here. Do we need `waitForAllContainerRemoval`?

Reply:
Did thorough tests, yest we need it. I was getting flaky tests.